src/core/execute.c

```diff
@@ -2576,12 +2583,6 @@ static int exec_child(
                 }
         }
 
-        r = apply_working_directory(context, params, home, needs_mount_namespace);
-        if (r < 0) {
-                *exit_status = EXIT_CHROOT;
-                return r;
-        }
-
```

Comment:
The why is try to move namespace operations, chdir, chroot etc close to each other, and delay a bit the group enforcing which should be one of the first operations of applying sandbox or switch the security context... Now from a practical view this is not really necessary since we should still have the capabilities to succeed at chdir() to any dir regardless of the gids... So yes consider it cleaning.